static/src/javascripts/lib/timeout.spec.js

```diff
@@ -0,0 +1,60 @@
+// @flow
+
+import timeout from 'lib/timeout';
+
+function wait(time) {
+    return new Promise(resolve => {
+        setTimeout(resolve, time);
+    });
+}
+
+describe('timeout', () => {
+    test(
+        "resolves with the promised value if the timeout doesn't fire",
+        done => {
+            timeout(10, Promise.resolve(12))
```

Comment:
I wonder if we could use [`chance`](http://chancejs.com/) (installed in #16156) to take care of these magic numbers?